### DISCUSSION TITLE
Required additional metadata

### DISCUSSION THREAD

**Discussion by jspaezp** [ProteomicsML/ProteomicsML#9] [General]
Hello there!

First of all, I love the idea of the repo and hope to contribute to it in the future!
I wanted to pitch the idea of having a more detailed metadata information.

In particular I think it would be great to specify:

1. For Retention time:
    1. Is the measured time peak apex or fragmentation?
    2. When calibrated to iRT how was it calculated?
    3. De-duplicated? Averaged? Range reported?
2. For spectra:
    1. Are the spectra raw or encoded?
    2. Is it a spectral archive or a spectral library (only identified vs non identified).
    4. Is the database redundant or has it been de-duplicated somehow?
        1. If de-duplicated, has it been averaged or "best peak" picked.
    5. Instrument and ideally instrument settings
3. For IMS (I am not that familiar with this data type):
    1. Instrument
    2. (I am not sure if it would apply but de-duplication as well)

Finally, what do you think about making the attributes section a TOML/YAML, it would still render well in the viewer but it would make it a lot more machine parsable.

LMK what you think
Best!
Sebastian

unrelated: is there a reason why the .qmd documents are rendered to jupyter? I really feel like the format is not great, if we want more version control, we could render to .md and if we want better display, we could render to html ... I am sorry if I have stronger opinions about the ipynb format than most people

**Comment by RobbinBouwmeester** [comment]
Dear Sebastian,

Thank you for your interest in ProteomicsML, and that you share our enthousiasm for the project ;).

> For Retention time:
>
>    Is the measured time peak apex or fragmentation?

Agreed, we should report where the retention time was measured, maybe it would even be a good idea to mention what software was used to determine the apex. The software can make quite a difference.

>    When calibrated to iRT how was it calculated?

Sure. I do think this is lower on the priority list than other points.

>    De-duplicated? Averaged? Range reported?

Agreed. This information should be included.

> For IMS (I am not that familiar with this data type):
>
>   Instrument

Sure, usually the type of ion mobility is more important though, but downstream data acquisition can have an effect on reliability on down-stream data.

>   (I am not sure if it would apply but de-duplication as well)

Agreed. This information should be included.

I refrained from answering the questions regarding fragmentation spectra, since I am not the one with the most expertise in this group regarding that data.

We will discuss these points soon. Formalizing the requirements for data sets is on our todo list.

Kind regards,

Robbin

**Comment by jspaezp** [comment]
thanks for replying!
let me know if you want to "crowdsource" any of the tasks/looking for the metadata in the papers.

best!
Sebastian

**Comment by RalfG** [comment]
Hi Sebastian,

Thanks for your feedback! 

With regard to metadata for fragmentation: I fully agree on all suggestions. We are currently setting up a list of "required" and "recommended" metadata items for each data type, and I will add yours to the list. 

> Finally, what do you think about making the attributes section a TOML/YAML, it would still render well in the viewer but it would make it a lot more machine parsable.

I agree, machine-readability would  definitly be a plus. However, to facilitate contributions from the community, we would like to keep the setup of the source markdown files as uncomplicated as possible.  Finding a good balance between those two aspects will require some thinking and investigation into the possibilities within the Quarto framework.

> unrelated: is there a reason why the .qmd documents are rendered to jupyter? I really feel like the format is not great, if we want more version control, we could render to .md and if we want better display, we could render to html ... I am sorry if I have stronger opinions about the ipynb format than most people

In the setup that we went for, with Quarto as static site rendering framework, all Jupyter, Markdown, and QMarkdown files are automatically rendered to html ([proteomicsml.github.io](https://github.com/ProteomicsML/proteomicsml.github.io)), which are then hosted on the [proteomicsml.org](https://www.proteomicsml.org) website. I think you asked this before the website was online? With regard to ipynb files and git, I am planning to implement some nbdev automations to reduce merging issues, large diffs, etc. With all its shortcomings, Jupyter does seem like the best system for the tutorials, especially with the links to Colab.

If you have more suggestions or ideas, let us know!

Best,
Ralf

**Reply by jspaezp** [reply]
ohhh I see, colab compatibility is definitely a plus
that is great!
